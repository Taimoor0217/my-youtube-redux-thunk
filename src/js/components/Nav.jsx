import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import Search from './Search';
import { getVideos } from '../actions';

const Nav = (props) => {
  return (
    <div className="nav">
      <Search {...props} />
    </div>
  )
}

Nav.propTypes = {
  getVideos: PropTypes.func.isRequired,
}

export default connect(null, {getVideos})(Nav);