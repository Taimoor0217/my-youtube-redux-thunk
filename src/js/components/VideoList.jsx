import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import VideoListEntry from './VideoListEntry';
import { selectVideo } from '../actions';

class VideoList extends Component {

  createList() {
    let { videoList, selectVideo } = this.props;
    return videoList.map((item, i) => {
      return <VideoListEntry key={i} video={item} handleSelectedVideo={selectVideo} />
    })
  }

  render() {
    let { videoList, isLoading, error } = this.props;
    return (
      <div className="videoList">
        {
          isLoading ? <div>Loding Videos....</div> :
          error ? <div>{error.message}</div> : this.createList()
        }
      </div>
    );
  }
}

VideoList.propTypes = {
  videoList: PropTypes.array.isRequired,
  selectVideo: PropTypes.func.isRequired,
}

function mapStateToProps({videoList}) {
  return {
    videoList: videoList.videos,
    isLoading: videoList.isLoading,
    error: videoList.error
  }
};

export default connect(mapStateToProps, {selectVideo})(VideoList);