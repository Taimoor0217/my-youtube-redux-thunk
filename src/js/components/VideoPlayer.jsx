import React, { Component } from 'react';
import PropTypes from 'prop-types';


const VideoPlayer = ({ seletedVideo }) => {
  if(Object.keys(seletedVideo).length) {
    var seletedVideoURL = "https://www.youtube.com/embed/" + seletedVideo.id.videoId
  }
  return (
    <div className="videoPlayer">
      <div className="player embed-responsive embed-responsive-16by9">
        <iframe 
          className="embed-responsive-item"
          src={selectedVideoURL} 
          frameborder="0"
          allowFullScreen
        />
      </div>
      <div className="video-details">
        <h4 className="video-title">
          {Object.keys(selectedVideo).length ? selectedVideo.snippet.title : ""}
        </h4>
        <p className="video-descriptions">
          {Object.keys(seletedVideo).length ? selectedVideo.snippet.description : ""}
        </p>
      </div>
    </div>
  )
};

VideoPlayer.propTypes = {
}

export default VideoPlayer;