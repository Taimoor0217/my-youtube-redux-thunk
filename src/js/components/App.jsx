import React, { Component } from 'react';
import { connect } from 'react-redux';

import Nav from './Nav';
import VideoPlayer from './VideoPlayer';
import VideoList from './VideoList';
import { getVideos } from '../actions';

class App extends Component {
  render() {
    return (
      <div id="app">
        <Nav />
        <div className="col-md-7">
          <VideoPlayer />
        </div>
        <div className="col-md-5">
          <VideoList />
        </div>
      </div>
    );
  }

  componentDidMount() {
    const { getVideos } = this.props;
    getVideos('javascript', true);
  }
}

export default connect(null, {getVideos})(App);