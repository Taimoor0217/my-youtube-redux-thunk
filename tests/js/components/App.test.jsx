import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { App } from '../../../src/js/components/App';

Enzyme.configure({ adapter: new Adapter() });

describe('testing App component', () => {
  const getVideos = jest.fn();
  const wrapper = shallow(<App getVideos={getVideos} />);
  it('renders the App component', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('tests that getVideos run with correct args', () => {
    expect(getVideos.mock.calls[0]).toEqual(['javascript', true]);
  });
});