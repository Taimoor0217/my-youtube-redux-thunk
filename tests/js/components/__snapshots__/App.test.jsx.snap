// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders 1`] = `
<div
  id="app"
>
  <Connect(Nav) />
  <div
    className="col-md-7"
  >
    <Connect(VideoPlayer) />
  </div>
  <div
    className="col-md-5"
  >
    <Connect(VideoList) />
  </div>
</div>
`;
